packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,232 @@
+            ? `${firstName.charAt(0)}${lastName.charAt(0)}`
+            : firstName.charAt(0);
+
+        return size === "xs" && letters.length > 1 ? letters.charAt(0) : letters;
+    }, [name, size]);
+
+    return (
+        <Text
+            size={AvatarToTextSizeAdapter[size]}
+            className={styles["hop-Avatar__initials"]}
+        >
+            {initials}
+        </Text>
+    );
+}
+
+function Avatar(props: AvatarProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, AvatarContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        "aria-label": ariaLabel,
+        className,
+        fallbackSrc,
+        imageProps,
+        isDisabled,
+        name,
+        size: sizeValue,
+        slot,
+        src,
+        style,
+        ...otherProps
+    } = ownProps;
+    const domProps = filterDOMProps(otherProps);
+    
+    const { onError, onLoad, ...otherImageProps } = imageProps ?? {};
+    const [imageUrl, status] = useImageFallback({ src, fallbackSrc, onError, onLoad });
+    const imageLoaded = status === "loaded";
+    const imageFailed = status === "failed";
+
+    const size = useResponsiveValue(sizeValue) ?? "md";
+    const isBrokenImage = src && imageFailed && fallbackSrc !== null;
+    const isImage = src && !imageFailed;
+    const isInitials = !src || (src && imageFailed && fallbackSrc === null);
+
+    const classNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            size,
+            isBrokenImage && "broken-image",
+            isImage && "image",
+            isInitials && getColorName(name)
+        ),
+        stylingProps.className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...stylingProps.style,
+        ...style
+    };
+
+    const handleError: ReactEventHandler<HTMLImageElement> = e => {
+        onError?.(e);
+    };
+
+    let content = <AvatarInitials
+        {...otherProps}
+        name={name}
+        size={size}
+    />;
+    
+    if (imageLoaded) {
+        content = <img
+            {...filterDOMProps(otherImageProps)}
+            alt=""
+            aria-hidden
+            className={styles["hop-Avatar__image"]}
+            onError={handleError}
+            src={imageUrl}
+        />;
+    }
+
+    if (imageFailed) {
+        if (fallbackSrc !== null) {
+            return (
+                <RichIconAvatarImage
+                    {...domProps}
+                    aria-label={ariaLabel ?? name}
+                    className={classNames}
+                    isDisabled={isDisabled}
+                    ref={ref}
+                    size={size}
+                    slot={slot}
+                    style={mergedStyles}
+                >
+                    <BrokenImageRichIcon />
+                </RichIconAvatarImage>
+            );
+        }
+    }
+
+    if (!isInitials && !imageLoaded) {
```

Comment:
can returning null while the image is loading cause UI jumps? like if while it's loading, the size is not reserved, then it loads, all the content will be pushed down. Should it be content = null, soo at least the div with the reserved size is rendered?